### DISCUSSION TITLE
Or query, with many parameters, does it require writing each one twice

### DISCUSSION THREAD

**Discussion by xuyongweb** [ujjwalguptaofficial/JsStore#330] [Q&A]
`select({
    from: 'Orders',
    where: [{
        customerId: 2,
        employeeId: 7,
       type: 1,
       delete: false
    }, {
        or: {
            customerId: 7,
            employeeId: 2,
           type: 1,
          delete: false
        }
    }]
})`

If there are other conditional queries, should we write two copies in both the outer layer and the or, for example: delete: false, type: 1

**Comment by xuyongweb** [comment]
Can the same conditions be presented separately

**Reply by ujjwalguptaofficial** [reply]
I am not sure what you mean by that - but let me try to explain.

Each `object` query in array represent & between them,  when using with `or` it becomes or (customerId;7 && employeeId:2) .

e.g - this query in sql 

```
Select * from Table_Name where Column1=value1 or (Column2=value2 and Column3=value3);
```

is equal to 

```
connection.select({
    from: "Table_Name",
    where: [{
            Column1: value1
        },
        {
            or: {
                Column2: value2,
                Column3: value3
            }
        }
    ]
});

```

I hope this helps you but let me know if you are not able to understand. It will be better if you will share the sql query and i share you the respective jsstore query.

**Comment by xuyongweb** [comment]
Select * from Table_Name where (Column1=value1 and Column2=value2 and Column3 = value3 and Column4 = value4)  or (Column2=value1 and Column1=value2 and Column3 = value3 and Column4 = value4);

 where Column3 = value3 and Column4 = value4 and ((Column2=value1 and Column1 = value2) or (Column1=value1 and Column2=value2))

May I ask how to write it using jsstore syntax?

**Comment by xuyongweb** [comment]
Select * from Table_Name where Column1=value1 and (Column2=value2 or Column3=value3);
@ujjwalguptaofficial
